CharlieBackend.Business/Services/LessonService.cs

```diff
@@ -39,6 +40,35 @@ public async Task<Result<LessonDto>> CreateLessonAsync(CreateLessonDto lessonDto
                 {
                     createdLessonEntity.Theme = foundTheme;
                 }
+                
+                var foundMentor = await _unitOfWork.MentorRepository.GetMentorByIdAsync(lessonDto.MentorId);
+
+                if (foundMentor == null)
+                {
+                    return Result<LessonDto>.GetError(ErrorCode.NotFound, "Mentor is not found"); // return Result<T>
+                }
+
+                var foundStudentGroup = await _unitOfWork.StudentGroupRepository.GetByIdAsync(lessonDto.StudentGroupId);
+
+                if (foundStudentGroup == null)
+                {
+                    return Result<LessonDto>.GetError(ErrorCode.NotFound, "Student group is not found");
+                }
+
+                if (foundStudentGroup.StudentsOfStudentGroups.Count != lessonDto.LessonVisits.Count)
```

Comment:
Why this parameter should be equal?
Not all of group members could be at a lesson.

Reply:
That's correct, but for each student in group should be created visit, either with Presence= 0 or 1